lib/active_record_shards/tasks.rb

```diff
@@ -12,7 +12,8 @@
       if key.starts_with?(env_name) && !key.ends_with?("_slave")
         begin
           if ActiveRecord::VERSION::MAJOR >= 4
-            ActiveRecord::Tasks::DatabaseTasks.drop(conf)
+            connection = ActiveRecord::Base.mysql2_connection(conf.merge('database' => nil))
```

Comment:
I spent a long time trying to figure out why connection() wasn't working. It seemed that even if I called establish_connection() beforehand, calling a subsequent connection() went back through the process of retrieving a connection from the connection_pool and then seemed to always be forced back to either the main DB or default shard connection, which would fail if either of those didn't exist.